amethyst_animation/src/systems/control.rs

```diff
@@ -439,13 +439,15 @@ where
     T: AnimationSampling + Component + Clone,
 {
     // check that hierarchy is valid, and all samplers exist
-    if animation
+    let still_loading = animation
         .nodes
         .iter()
         .any(|&(ref node_index, _, ref sampler_handle)| {
             !hierarchy.nodes.contains_key(node_index)
                 || sampler_storage.get(sampler_handle).is_none()
-        }) {
+        });
+
+    if still_loading {
```

Comment:
What's the motivation for this change?

Reply:
The idea was to take the "complex" if statement and introduce an intermediate variable name that represents the intention of the statement. I assume that the comment above the statement could be seen as doing the same as the its only the check for a precondition.

Reply:
Restored the original version.